src/main/kotlin/no/nav/helse/spion/vedtaksmelding/RecurringJob.kt

```diff
@@ -0,0 +1,53 @@
+package no.nav.helse.spion.vedtaksmelding
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.time.delay
+import org.slf4j.LoggerFactory
+import java.time.Duration
+
+abstract class RecurringJob(
+        private val coroutineScope: CoroutineScope,
+        private val waitTimeBetweenRuns: Duration) {
+
+    protected val logger = LoggerFactory.getLogger(this::class.java)
+
+    private var isRunning = false
+
+    fun startAsync(retryOnFail: Boolean = false) {
```

Comment:
Skulle retry on fail vært knyttet til et antall ganger? Eller?

Reply:
Nei jeg tenker at enten vil vi at den skal prøve til det går, eller så bare lar vi kubernetes ta seg av det. Lar den i alle fall ligge for nå. Å støtte de 2 modusene var mest et innfall :)